src/main/java/io/anserini/index/IndexCollection.java

```diff
@@ -656,10 +656,12 @@ public void run() throws IOException {
     if (indexPath != null && !args.dryRun) {
 
       final Directory dir = FSDirectory.open(indexPath);
+      final CJKAnalyzer chineseAnalyzer = new CJKAnalyzer();
       final EnglishStemmingAnalyzer analyzer = args.keepStopwords ?
           new EnglishStemmingAnalyzer(args.stemmer, CharArraySet.EMPTY_SET) : new EnglishStemmingAnalyzer(args.stemmer);
       final TweetAnalyzer tweetAnalyzer = new TweetAnalyzer(args.tweetStemming);
-      final IndexWriterConfig config = args.collectionClass.equals("TweetCollection") ? new IndexWriterConfig(tweetAnalyzer) : new IndexWriterConfig(analyzer);
+      final IndexWriterConfig config = args.collectionClass.equals("TweetCollection") ?
+          new IndexWriterConfig(tweetAnalyzer) : args.language.equals("zh") ? new IndexWriterConfig(chineseAnalyzer) : new IndexWriterConfig(analyzer);
```

Comment:
Maybe this should be a normal nested if statement to improve readability?

Reply:
+1